### PR TITLE
pythonPackages.capstone: 3.0.5.post1 -> 4.0.1, redesign as wrapper package around main capstone package

### PR DESCRIPTION
**Pull request by risicle**
##### Motivation for this change
Similar to #76762, this allows us to more easily keep the main package and python bindings in sync.

Again, I've also enabled the main `capstone` package on darwin and enabled its tests too.

Downstream dependency `pwntools` fails to build but that's due to `tox` being broken.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @thoughtpolice @bennofs

### CODE REVIEW

File: pkgs/development/libraries/capstone/default.nix [hunk 1]
```diff
@@ -23,7 +37,7 @@ stdenv.mkDerivation rec {
     description = "Advanced disassembly library";
     homepage    = "http://www.capstone-engine.org";
     license     = stdenv.lib.licenses.bsd3;
-    platforms   = stdenv.lib.platforms.linux;
-    maintainers = [ stdenv.lib.maintainers.thoughtpolice ];
+    platforms   = stdenv.lib.platforms.unix;
+    maintainers = with stdenv.lib.maintaners; [ thoughtpolice ris ];
```
Comment (jonringer) [inline pkgs/development/libraries/capstone/default.nix]:
```suggestion
    maintainers = with stdenv.lib.maintainers; [ thoughtpolice ris ];
```

Comment (risicle) [reply pkgs/development/libraries/capstone/default.nix]: Damnit. Fixed.

**Review by thoughtpolice — APPROVED** [review]
Excellent, thank you for this! Very useful.

**Comment by thoughtpolice** [comment]
@GrahamcOfBorg build capstone pythonPackages.capstone

**Comment by risicle** [comment]
Pushed an extra commit fixing the `pkg-config` output. Turns out it generates `capstone.pc` during the `buildPhase` and therefore needs `PREFIX` to get it right.

**Review by jonringer — APPROVED** [review]
diff LGTM

```
16 package built:
boomerang capstone pwndbg python27Packages.ROPGadget python27Packages.capstone python27Packages.pwntools python27Packages.ropper python37Packages.ROPGadget python37Packages.capstone python37Packages.pwntools python37Packages.ropper python38Packages.ROPGadget python38Packages.capstone python38Packages.pwntools python38Packages.ropper wcc
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build boomerang capstone pwndbg python27Packages.ROPGadget python27Packages.capstone python27Packages.pwntools python27Packages.ropper python37Packages.ROPGadget python37Packages.capstone python37Packages.pwntools python37Packages.ropper python38Packages.ROPGadget python38Packages.capstone python38Packages.pwntools python38Packages.ropper wcc

**Review by jonringer — COMMENTED** [review]
I think this should be squashed into two commits, one for capstone, one for the python package

**Comment by risicle** [comment]
I can do that I was just trying to give people more points to bisect to if this screws things up - it also allows me to comment things separately...

**Review by jonringer — APPROVED** [review]
I can see it either way, personally I view it as "gave darwin some love while bumping the version".

cc @FRidh

**Comment by risicle** [comment]
Sure, done.

**Comment by jonringer** [comment]
@GrahamcOfBorg build boomerang capstone pwndbg python27Packages.ROPGadget python27Packages.capstone python27Packages.pwntools python27Packages.ropper python37Packages.ROPGadget python37Packages.capstone python37Packages.pwntools python37Packages.ropper python38Packages.ROPGadget python38Packages.capstone python38Packages.pwntools python38Packages.ropper wcc

**Pull request merged**